R/training_window.R

```diff
@@ -0,0 +1,109 @@
+#' Limits the size of the training window to the most recent observations
+#'
+#' `step_training_window` creates a *specification* of a recipe step that
+#'   limit the size of the training window to the `nrec` most recent
+#'   observations in `time_value` per location from `geo_value`.
+#'
+#' @param recipe A recipe object. The step will be added to the
+#'  sequence of operations for this recipe.
+#' @param ... One or more selector functions to choose variables
```

Comment:
Do we actually need this argument? I don't think we need `role` either.

Basically, this operates on the `time_value` (and any keys) column right? It doesn't need anything else?

Reply:
It's probably good to make this resemble [`step_slice()`](https://recipes.tidymodels.org/reference/step_slice.html), since that's basically what we're doing.

Reply:
So, I'm thinking just say that `role` and `...` are unused.

Reply:
Ok. I removed `...` with no complaints from RStudio, But when I remove `role`, I get an error after using prep that says "Error in if (!is.na(x$steps[[i]]$role)) { : argument is of length zero". Based on that, I think we should keep role even if it isn't used. Also, it is kept in `step_slice()`, so I think it is probably ok to also keep here. I'll just note in the documentation that it is unused, as you said.

Reply:
And to address your 2nd q. from your first comment on this code, the fun. needs `time_value` and `geo_value` columns as the issue said to limit the training window n most recent observations per location.